utf8.h

```diff
@@ -662,6 +665,54 @@ void* utf8str(const void* haystack, const void* needle) {
   return 0;
 }
 
+void* utf8casestr(const void* haystack, const void* needle) {
+	const char* h = (const char*)haystack;
+
+	// if needle has no utf8 codepoints before the null terminating
+	// byte then return haystack
+	if ('\0' == *((const char*)needle)) {
+		return (void*)haystack;
+	}
+
+	while ('\0' != *h) {
+		const char* maybeMatch = h;
+		const char* n = (const char*)needle;
+
+		while (true) {
```

Comment:
Looks like this is causing the build error. OpenBoardView is C++, hence why I didn't notice